extensions-core/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsDataSegmentUtil.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.storage.hdfs;
+
+import com.google.common.base.Throwables;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import java.io.IOException;
+
+
+public class HdfsDataSegmentUtil
+{
+  /**
+   * returns fully qualified hdfs path string.
+   */
+  public static String getFullyQualifiedHdfsPath(String path, Configuration config)
+  {
+    if (new Path(path).isAbsoluteAndSchemeAuthorityNull()) {
+      try {
+        FileSystem localHdfsFileSystem = FileSystem.newInstance(config);
+        return new Path(
+            localHdfsFileSystem.getUri().getScheme(),
+            localHdfsFileSystem.getUri().getAuthority(),
+            path
+        ).toUri().getPath();
+      }
+      catch (IOException exception) {
+        Throwables.propagate(exception);
```

Comment:
if this Function is needed (see comment above)  also you don't want to propagate the `IOException` it will better give the user chance to handle it by having it as a checked exception